example/main.cpp

```diff
@@ -1,24 +1,139 @@
 #include <iostream>
 #include <threadpool.hpp>
+#include <random>
+#include <condition_variable>
 
 using namespace maxtek;
 
-void trivialFunc()
+class consumerProducerExample
 {
-    while (1)
-    {
-      std::cout << "Hello from task\n";
-    }
-}
+    public:
+        consumerProducerExample(size_t num_consumer_threads = std::thread::hardware_concurrency(),
+        size_t num_producer_threads = std::thread::hardware_concurrency(),
+        size_t buffer_size = 10, int consumer_thread_sleep = 100,
+        int producer_thread_sleep = 100): num_consumer_threads(num_consumer_threads),
+        num_producer_threads(num_producer_threads), buffer_size(buffer_size),
+        consumer_thread_sleep(consumer_thread_sleep),
+        producer_thread_sleep(producer_thread_sleep)
+        {
+            item_buf.reserve(buffer_size);
+            producer_futures.reserve(num_producer_threads);
+            myThreadPool = new threadpool(num_producer_threads + num_consumer_threads);
```

Comment:
As mentioned with the `shared_ptr` or `unique_ptr`, C++ gives us access to other functions for constructing objects. Use `make_shared` or `make_unique`.

[C++ docs for make_shared](https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared)